proposal/more-math-functions.md

```diff
@@ -0,0 +1,445 @@
+
+* Throw error if the units of `$dividend` and `$divisor` are not
+  [compatible](#compatible).
+* Return `$dividend` / `$divisor`.
+
+#### `mod()`
+
+```
+mod($dividend, $modulus)
+```
+
+* Throw error if the units of `$dividend` and `$modulus` are not
+  [compatible](#compatible).
+* Return `$dividend` % `$modulus`.
+
+### Exponentiation
+
+#### `log()`
+
+```
+log($number)
+```
+* Throw error if `$number` has units or `$number` < 0.
+* Return -Infinity if `$number` is 0.
+* Return Infinity if `$number` is infinite.
+* Return the natural log of `$number`.
+
+```
+log($number, $base)
+```
+
+* Throw error if `$base` < 0 or `$base` is 0 or `$base` is 1.
+* Return [log](#log)(`$number`) / [log](#log)(`$base`).
+
+#### `pow()`
+
+```
+pow($base, $exponent)
+```
+
+* Throw error if `$base` or `$exponent` has units.
+
+* If `$exponent` is 0:
+  * Return 1.
+
+* Else if `$exponent` is infinite:
+  * Return NaN if [abs](#abs)(`$base`) is 1.
+  * Return Infinity if [abs](#abs)(`$base`) > 1 and `$exponent` > 0, or if
+    [abs](#abs)(`$base`) < 1 and `$exponent` < 0.
+  * Return 0.
+
+* Else if `$exponent` is finite:
+  * Return NaN if `$base` < 0 and `$exponent` is not an integer.
+
+  * Return Infinity if `$base` is 0 and `$exponent` < 0, or if `$base` is
+    Infinity and `$exponent` > 0.
+  * If `$base` is -0 and `$exponent` < 0, or if `$base` is -Infinity and
+    `$exponent` > 0:
+    * Return -Infinity if `$exponent` is an odd integer.
+    * Return Infinity.
+
+  * Return 0 if `$base` is 0 and `$exponent` > 0, or if `$base` is Infinity and
+    `$exponent` < 0.
+  * If `$base` is -0 and `$exponent` > 0, or if `$base` is -Infinity and
+    `$exponent` < 0:
+    * Return -0 if `$exponent` is an odd integer.
+    * Return 0.
+
+  * Return `$base` raised to the power of `$exponent`.
+
+#### `sqrt()`
+
+```
+sqrt($number)
+```
+
+* Throw error if `$number` has units.
+* Return NaN if `$number` < 0.
+* Return -0 if `$number` is -0.
+* Return Infinity if `$number` is infinite.
+* Return the square root of `$number`.
+
+### Trigonometry
+
+#### Procedures
+
+##### Verify Inputs
+
+`cos($number), sin($number), and tan($number)` throw error if `$number` has
+units but is not a rotation value. If `$number` is unitless, it is assumed to be
+in radians.
+
+`acos($number), asin($number), and atan($number)` throw error if `$number` has
+units.
+
+Both inputs to `atan2($y, $x)` must have the same unit or be unitless. If not,
+throw error.
+
+#### `cos()`
+
+```
+cos($number)
+```
+
+* [Verify inputs](#verify-inputs).
+* Return NaN if `$number` is infinite.
+* Return cos(`$number`).
+
+#### `sin()`
+
+```
+sin($number)
+```
+
+* [Verify inputs](#verify-inputs).
+* Return NaN if `$number` is infinite.
+* Return -0 if `$number` is -0.
+* Return sin(`$number`).
+
+#### `tan()`
+
+```
+tan($number)
+```
+
+* [Verify inputs](#verify-inputs).
+* Return NaN if `$number` is infinite.
+* Return -0 if `$number` is -0.
+* If `$number` is an asymptote value:
```

Comment:
Define "asymptote value"

Reply:
I realized the word "asymptote" is kind of a red herring, I just removed this line to let the logic in the next two lines speak for themselves.